var/spack/repos/builtin/packages/py-ctranslate2/package.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyCtranslate2(CMakePackage, PythonExtension, CudaPackage):
+    """CTranslate2 is a C++ and Python library for efficient inference with
+    Transformer models."""
+
+    homepage = "https://github.com/OpenNMT/CTranslate2"
+    git = "https://github.com/OpenNMT/CTranslate2.git"
+
+    maintainers("meyersbs", "aweits")
+
+    version("3.12.0", tag="v3.12.0", submodules=True)
+
+    extends("python")
+
+    # From setup.py, CMakeLists.txt, requirements.txt
+    depends_on("py-setuptools", type="build")
+    depends_on("cmake@3.15:", type="build")
+    depends_on("py-pip", type="build")
+    depends_on("py-pybind11", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-pyyaml@5.3:6", type=("build", "run"))
+
+    # From build errors
+    depends_on("py-intel-openmp", type=("build", "run"))
+    # From https://opennmt.net/CTranslate2/installation.html#build-options
+    variant("mkl", default=True, description="Build with mkl")
+    variant("cuda", default=False, description="Build with cuda")
+    variant("cudnn", default=False, description="Build with cudnn")
+    depends_on("intel-mkl@2019.5:", when="+mkl", type=("build", "run"))
+    depends_on("cuda@11.0:", when="+cuda", type=("build", "run"))
+    depends_on("cudnn@8:", when="+cudnn", type=("build", "run"))
+
+    def setup_build_environment(self, env):
+        # https://opennmt.net/CTranslate2/installation.html#compile-the-python-wrapper
+        env.append_path("SPACK_INCLUDE_DIRS", prefix.include)
+        env.append_path("SPACK_LINK_DIRS", prefix.lib)
+        env.append_path("SPACK_LINK_DIRS", prefix.lib64)
+        env.set("CTRANSLATE2_ROOT", self.spec.prefix)
+        env.prepend_path("LD_LIBRARY_PATH", self.spec.prefix)
+
+    def setup_run_environment(self, env):
+        # https://opennmt.net/CTranslate2/installation.html#compile-the-python-wrapper
+        env.prepend_path("LD_LIBRARY_PATH", self.spec.prefix)
```

Comment:
```suggestion
        env.prepend_path("LD_LIBRARY_PATH", self.spec["py-ctranslate2"].libs.directories[0])
```
The libs are probably in `<prefix>/lib`, not `<prefix>`. You'll have to check if the libs attribute returns the correct thing. If not, you can override it in the package.

Reply:
With that change, I get this error: 

```
==> Error: NoLibrariesError: Unable to recursively locate py-ctranslate2 libraries in /home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-ctranslate2-3.12.0-6zrs4uc2v7pxnf4m
gkfhdnp2finhe2bs
```

Reply:
Then you'll have to override the `libs` attribute. I'm not sure what the library name is or where it is located.

Reply:
I don't know what's changed since May 16th, but I tested without doing anything in `setup_run_environment` and everything seems to be working... @qwertos is as confused as I am